File: src/main/java/ru/translater/generate/GenerateProperties.java
package ru.translater.generate;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import ru.translater.common.CommonProperties;

@Data
@Configuration
@ConfigurationProperties(prefix = "generate")
public class GenerateProperties extends CommonProperties {
    private String formatStorePath;
    private String htmlTemplate;
    private String imagePath;
    private String configTemplatePath;
}

File: src/main/kotlin/com/translater/generate/properties/GenerateProperties.kt
package com.translater.generate.properties

import com.translater.common.properties.CommonProperties
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(prefix = "generate")
data class GenerateProperties(
    override val storePath: String,
    val formatStorePath: String,
    val imagePath: String,
    val configTemplatePath: String
) : CommonProperties(storePath)